dbms/src/Operators/ExchangeSenderSinkOp.cpp

```diff
@@ -28,15 +28,34 @@ void ExchangeSenderSinkOp::operateSuffixImpl()
 
 OperatorStatus ExchangeSenderSinkOp::writeImpl(Block && block)
 {
-    if (!block)
+    assert(!buffer);
+    buffer.emplace(std::move(block));
+    return tryFlush();
+}
+
+OperatorStatus ExchangeSenderSinkOp::tryFlush()
+{
+    assert(buffer);
+    auto res = waitForWriter();
```

Comment:
Perhaps moving `waitForWriter` into `Writer::write` and `Writer::flush` would be better. This way, even if the tunnel is busy, write can still batch in memory.

Reply:
You mean if the block's size is small, it will be cached in the writer without writting to tunnel?

Reply:
yes, if the total block rows is smaller than `batch_send_min_limit`, MPPTunnel::write would not be call.
https://github.com/pingcap/tiflash/blob/master/dbms/src/Flash/Mpp/BroadcastOrPassThroughWriter.cpp#L103-L109

Reply:
Good idea, I reimplement this pr by moving `waitForWriter` to `Writer::flush`. But `waitForWriter` is still needed in `ExchangeSenderSinkOp::await`

Reply:
lgtm